seed/tests/test_derived_columns.py

```diff
@@ -0,0 +1,611 @@
+
+    :param operators: list[str], list of operators to choose from
+    :param operand_st: Strategy, for selecting values to use in operations
+    :max_operands: int, maximum number of operands
+    :return: str
+    """
+    operands = draw(
+        st.lists(
+            operand_st,
+            min_size=2,
+            max_size=max_operands
+        )
+    )
+
+    n_operators = len(operands) - 1
+    operators = draw(
+        st.lists(
+            st.sampled_from(operators),
+            min_size=n_operators,
+            max_size=n_operators
+        )
+    )
+
+    result = ''
+    # insert an operator between each operand
+    for idx, operator in enumerate(operators):
+        result += f'{operands[idx]} {operators[idx]} '
+    result += str(operands[-1])
+
+    if draw(in_parens_st):
+        result = f'({result})'
+
+    return result
+
+
+@st.composite
+def parameter_name_st(draw):
+    """Strategy for creating valid parameter names. e.g. _hello, test123, etc
+
+    :return: str
+    """
+    prefix = draw(
+        st.text(ascii_letters + '_', min_size=1, max_size=1)
+    )
+    body = draw(
+        st.text(digits + ascii_letters + '_')
+    )
+    return f'{prefix}{body}'
+
+
+# atomic strategies (excluding parameter names)
+atomic_no_params_st = good_floats.map(str) \
+    | pow_st() \
+    | func_st(st.sampled_from(['min', 'max']), min_args=2) \
+    | func_st(st.just('abs'), min_args=1, max_args=1)
+
+# function for recursive strategy
+# avoiding the division operator b/c it can cause unexpected divide by zero issues
+recursive_st_func = lambda children: \
+    arithmetic_st(operators=['+', '-', '*'], operand_st=children, max_operands=3, in_parens_st=st.booleans()) \
+    | func_st(st.sampled_from(['min', 'max']), children, min_args=2) \
+    | func_st(st.just('abs'), st.one_of(children), min_args=1, max_args=1)
+
+# strategy for generating complex / nested expressions WITHOUT parameters
+full_expression_no_params_st = st.recursive(atomic_no_params_st, recursive_st_func)
+
+
+@st.composite
+def full_expression_with_params_st(draw, parameters=st.dictionaries(parameter_name_st(), good_floats, min_size=1)):
+    """Strategy for generating complex / nested expressions that include parameters
+
+    :param parameters: Strategy, should provide dictionaries where keys are valid
+        python var names and values are numeric
+    :return: tuple(str, dict), the generated expression and parameters
+    """
+    params = draw(parameters)
+
+    template_params = [f'${param_name}' for param_name in params.keys()]
+    # NOTE: this won't ensure all (or even any) param names end up in the expression
+    # however our evaluator should be fine with us passing extra parameters
+    expression = draw(
+        st.recursive(
+            atomic_no_params_st | st.sampled_from(template_params),
+            recursive_st_func
+        )
+    )
+
+    return expression, params
+
+
+class TestExpressionEvaluator(TestCase):
+    @no_deadline
```

Comment:
@adrian-lara 
```suggestion
class TestExpressionEvaluator(TestCase):
    """NOTE:
    This collection of tests uses the hypothesis testing library
    See: https://hypothesis.readthedocs.io/en/latest/index.html

    Instead of making assertions about specific data, we tell hypothesis what the data _basically_
    looks like, and it runs our tests many times over different types of data it generates.
    This allows us to test cases we would have never though of and find edge cases.

    We create data specifications (ie what the data should look like) with "strategies".
    We have written custom strategies for generating different types of expressions above.
    They all end with the `_st` suffix.
    We then tell hypothesis which strategy to use for a test (ie what the data input is) by using the
    `@given` decorator.
    """

    @no_deadline
```